app/src/ui/diff/index.tsx

```diff
@@ -1044,52 +1122,31 @@ export class Diff extends React.Component<IDiffProps, {}> {
     this.codeMirror = cmh === null ? null : cmh.getEditor()
   }
 
-  public render() {
-    const diff = this.props.diff
-
-    if (diff.kind === DiffType.Image) {
-      return this.renderImage(diff)
-    }
-
-    if (diff.kind === DiffType.Binary) {
-      return this.renderBinaryFile()
-    }
-
-    if (diff.kind === DiffType.TooLarge) {
-      const BlankSlateImage = encodePathAsUrl(
-        __dirname,
-        'static/empty-no-file-selected.svg'
-      )
-      const diffSizeMB = Math.round(diff.length / (1024 * 1024))
-      return (
-        <div className="panel empty">
-          <img src={BlankSlateImage} className="blankslate-image" />
-          The diff returned by Git is {diffSizeMB}MB ({diff.length} bytes),
-          which is larger than what can be displayed in GitHub Desktop.
-        </div>
-      )
-    }
-
-    if (diff.kind === DiffType.Text) {
-      if (diff.hunks.length === 0) {
-        if (this.props.file.status === AppFileStatus.New) {
-          return <div className="panel empty">The file is empty</div>
+  private renderDiff(diff: IDiff): JSX.Element | null {
+    switch (diff.kind) {
+      case DiffType.Text:
+        return this.renderText(diff)
+      case DiffType.Binary:
+        return this.renderBinaryFile()
+      case DiffType.Image:
+        return this.renderImage(diff)
+      case DiffType.LargeText: {
+        const largeTextDiff = diff as ILargeTextDiff
+
+        if (largeTextDiff.hunks == null || largeTextDiff.text == null) {
```

Comment:
the only type we would end up in this case is if `!isValidBuffer(buffer) === true` right? bufferTooLarge and diffTooLarge both have hunks and text from `diffFromRawDiffOutput(buffer)` of type `IRawDiff` which has hunks and contents. 

If that's true, would it make sense to add a DiffType.InvalidBuffer as a separate case in the switch statement then to result in `this.renderDiffTooLarge()`?

Reply:
Yea, I like that because it's more explicit.